### ISSUE TITLE
Does this work with generic arrow functions?

### ISSUE THREAD

**Issue by jakeklassen** [antfu/esno#1]
I was testing this out with a quick sample file:

```typescript
// index.ts
console.log('🚀');

// Works
// function nada<T = unknown>() {
//   return {} as T;
// }

const nada = <T = unknown>() => ({} as T);
```

And received the following error when running `npx esno src/index.ts`:

```
/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:1
Error: Transform failed with 1 error:
/home/jake/workspace/typescript/typescript-node-base/src/index.ts:8:16: error: Expected ">" but found "="
    at failureErrorWithLog (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:458:15)
    at /home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:411:31
    at handleIncomingPacket (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:372:9)
    at readFromStdout (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:320:7)
    at runServiceSync (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:602:3)
    at Object.transformSync (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild/lib/main.js:506:3)
    at compile (/home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild-register/dist/node.js:55:51)
    at Object._compile (/home/jake/workspace/typescript/typescript-node-base/node_modules/pirates/lib/index.js:93:29)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10) {
  errors: [
    {
      location: {
        column: 16,
        file: '/home/jake/workspace/typescript/typescript-node-base/src/index.ts',
        length: 1,
        line: 8,
        lineText: 'const nada = <T = unknown>() => ({} as T);'
      },
      text: 'Expected ">" but found "="'
    }
  ],
  warnings: []
}
child_process.js:655
    throw err;
    ^

Error: Command failed: node -r /home/jake/workspace/typescript/typescript-node-base/node_modules/esm/esm.js -r /home/jake/workspace/typescript/typescript-node-base/node_modules/esbuild-register/register.js "src/index.ts"
    at checkExecSyncError (child_process.js:616:11)
    at execSync (child_process.js:652:15)
    at Object.<anonymous> (/home/jake/workspace/typescript/typescript-node-base/node_modules/esno/esno.js:13:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  status: 1,
  signal: null,
  output: [ null, null, null ],
  pid: 7144,
  stdout: null,
  stderr: null
}
```

The long form works fine (commented out), but the arrow function does not.

I'm not very familiar with esbuild yet, so this might just be ignorance on my part.

**Comment by antfu** [comment]
Yeah, probably a bug from ESBuild. Mind file an issue to it instead? Thanks.

**Comment by evanw** [comment]
This is a known problem with the `esbuild-register` package, not with esbuild itself: https://github.com/egoist/esbuild-register/issues/3. It incorrectly uses the `tsx` loader for `.ts` files instead the `ts` loader.

**Comment by antfu** [comment]
@jakeklassen fixed in v0.2.3 using a knightly build of https://github.com/egoist/esbuild-register/pull/5. Will switch back when it gets merged.
